services/test/regtest/regtest_client_test.go

```diff
@@ -180,6 +181,20 @@ func waitForPaymentCompletion(
 		castPayment, didCast = (*payment).(objects.LightningTransaction)
 		require.True(t, didCast)
 	}
+	if castPayment.GetStatus() == objects.TransactionStatusFailed {
+		if reflect.TypeOf(castPayment) == reflect.TypeOf(objects.OutgoingPayment{}) {
```

Comment:
Why do we have to cast to outgoingpayment here? I think this is also used for IncomingPayments. Might be good to use LightningTransaction like above.

Reply:
@jklein24 , because if using LightningTransaction, I was not able to get the FailureReason. And also IncomingPayment does not have that either.

Reply:
ahh ok that makes sense :-/